packages/promotion/src/initialize/index.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  ExternalModuleDeclaration,
+  InternalModuleDeclaration,
+  MedusaModule,
+  MODULE_PACKAGE_NAMES,
+  Modules,
+} from "@medusajs/modules-sdk"
+import { IPromotionModuleService, ModulesSdkTypes } from "@medusajs/types"
+import { moduleDefinition } from "../module-definition"
+import { InitializeModuleInjectableDependencies } from "../types"
+
+export const initialize = async (
+  options?:
+    | ModulesSdkTypes.ModuleServiceInitializeOptions
+    | ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
+    | ExternalModuleDeclaration
+    | InternalModuleDeclaration,
+  injectedDependencies?: InitializeModuleInjectableDependencies
+): Promise<IPromotionModuleService> => {
+  const serviceKey = Modules.PROMOTION
+
+  const loaded = await MedusaModule.bootstrap<IPromotionModuleService>({
+    moduleKey: serviceKey,
+    defaultPath: MODULE_PACKAGE_NAMES[Modules.PROMOTION],
+    declaration: options as
```

Comment:
question: Since we keep casting all the time in the initialize, would it make sense to change the declaration type of the bootstrap method to also allow for 
```ts
    | ModulesSdkTypes.ModuleServiceInitializeOptions
    | ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
```

Reply:
@adrien2p the bootstrap method isn't prepared for these 2 types, atleast on a type level. Adding that to the bootstrap method is breaking a chain of types all underneath it. I've added a TODO in the types to readd it when those are fixed, but this isn't in the scope of this PR.

Reply:
ok got it